controllers/controlplane/controller_util.go

```diff
@@ -60,30 +60,32 @@ func createNestedComponentSts(ctx context.Context,
 			"will use the default template.")
 		ncSts, err = genStatefulSetObject(ncMeta, ncSpec, ncKind, clusterName, cli, log)
 		if err != nil {
-			return fmt.Errorf("fail to generate the "+
-				"NestedEtcd Statefulset object: %v", err)
+			return fmt.Errorf("fail to generate the Statefulset object: %v", err)
 		}
 
-		ncSvc, err = genServiceObject(ncMeta, ncSpec, ncKind, clusterName, log)
-		if err != nil {
-			return fmt.Errorf("fail to generate the "+
-				"NesteEtcd Service object: %v", err)
+		if ncKind != clusterv1.ControllerManager {
+			// no need to create the service for the NestedControllerManager
```

Comment:
Just a note: For now I think this is true but if we want to run things like metrics service and be able to expose these we'd want the capability later.

Reply:
Yes, I agree. We can add it back once we need to access the controller manager